shop.go

```diff
@@ -91,6 +96,13 @@ func NewCustomShop(url string, accessToken string, isPlus bool) Shop {
 		return fmt.Sprintf("%v/%v", url, endpoint)
 	}
 
+	options := Options{}
+	for _, fn := range optionsFns {
+		fn(&options)
+	}
+
+	client.WithExponentialBackoff(options.retryCount, options.retryBaseDuration, options.retryMaxDuration)
```

Comment:
Idea was this would be another "option function", so it'd fit into the `http.NewClient` constructor above